packages/react-components/src/components/VideoGallery.test.tsx

```diff
@@ -0,0 +1,194 @@
+      createRemoteParticipant({
+        videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+      })
+    );
+    root.setProps({ remoteParticipants });
+    expect(tileCount(root)).toBe(11);
+    expect(audioTileCount(root)).toBe(11);
+    expect(videoTileCount(root)).toBe(0);
+    expect(gridTileCount(root)).toBe(11);
+    expect(gridAudioTileCount(root)).toBe(11);
+    expect(gridVideoTileCount(root)).toBe(0);
+    expect(root.find(HorizontalGallery).exists()).toBe(false);
+  });
+
+  test('should render max allowed video tiles with streams in the grid ', () => {
+    const localParticipant = createLocalParticipant({
+      videoStream: { isAvailable: true, renderElement: createVideoDivElement() }
+    });
+    // Need to mock hook _useContainerWidth because the returned width is used by HorizontalGallery to decide how many
+    // tiles to show
+    jest.spyOn(responsive, '_useContainerWidth').mockImplementation(() => {
+      return 500;
+    });
+
+    const root = mountVideoGalleryWithLocalParticipant({ localParticipant });
+
+    root.setProps({ layout: 'default' });
+    const remoteParticipants = Array.from({ length: 10 }, () =>
+      createRemoteParticipant({
+        videoStream: { isAvailable: true, renderElement: createVideoDivElement() }
+      })
+    );
+    root.setProps({ remoteParticipants });
+    expect(gridVideoTileCount(root)).toBe(DEFAULT_MAX_REMOTE_VIDEO_STREAMS + 1); // +1 for the local video stream
+    expect(root.find(HorizontalGallery).find(VideoTile).length).toBe(2);
+  });
+});
+
+const mountVideoGalleryWithLocalParticipant = (attrs: {
+  localParticipant: VideoGalleryLocalParticipant;
+}): ReactWrapper<VideoGalleryProps> => {
+  const { localParticipant } = attrs;
+  return mount(<VideoGallery localParticipant={localParticipant} />);
+};
+
+const tileCount = (root: ReactWrapper<VideoGalleryProps>): number => root.find(VideoTile).length;
+
+const videoTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(VideoTile).filterWhere((node) => {
+    return node.find(Persona).length === 0 && node.find(StreamMedia).length === 1;
+  }).length;
+
+const audioTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(VideoTile).filterWhere((node) => {
+    return node.find(Persona).length === 1 && node.find('video').length === 0;
+  }).length;
+
+const gridTileCount = (root: ReactWrapper<VideoGalleryProps>): number => root.find(GridLayout).find(VideoTile).length;
+
+const gridVideoTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(GridLayout).find(StreamMedia).length;
+
+const gridAudioTileCount = (root: ReactWrapper<VideoGalleryProps>): number =>
+  root.find(GridLayout).find(Persona).length;
+
+const createLocalParticipant = (attrs?: Partial<VideoGalleryLocalParticipant>): VideoGalleryLocalParticipant => {
+  return {
+    userId: attrs?.userId ?? `localParticipant-${Math.random()}`,
```

Comment:
Let's avoid randomness in tests as much as possible. There's only ever one local participant, right? Can we hard-code the default. The test can provided a non-default value if it wants for a second local participant.